File: Benchmarks/Gibberish.cs
using System;

namespace Benchmarks {
	/// <summary>
	/// Provides generators for gibberish.
	/// </summary>
	internal static class Gibberish {
		/// <summary>
		/// Generate gibberish.
		/// </summary>
		/// <param name="reductionFactor">The factor by which to reduce the total gibberish generated. Without a reduction factor it generates an <see cref="Array"/> of length <see cref="Int32.MaxValue"/>. The factor cuts this length down.</param>
		/// <returns>An <see cref="Array"/> of UTF-8 gibberish.</returns>
		public static Byte[] GenerateUtf8(Int32 reductionFactor) {
			Random random = new Random();
			Byte[] result = new Byte[Int32.MaxValue / reductionFactor];
			random.NextBytes(result);
			return result;
		}

		/// <summary>
		/// Generate gibberish.
		/// </summary>
		/// <param name="reductionFactor">The factor by which to reduce the total gibberish generated. Without a reduction factor it generates an <see cref="Array"/> of length <see cref="Int32.MaxValue"/>. The factor cuts this length down.</param>
		/// <returns>An <see cref="Array"/> of UTF-16 gibberish.</returns>
		public static Char[] GenerateUtf16(Int32 reductionFactor) {
			Random random = new Random();
			Char[] result = new Char[Int32.MaxValue / reductionFactor];
			for (Int32 i = 0; i < result.Length; i++) {
				result[i] = (Char)(random.Next() % UInt16.MaxValue);
			}
			return result;
		}
	}
}

File: Generators/Gibberish/Gibberish.vb
Imports System.Text

''' <summary>
''' Provides generators for gibberish
''' </summary>
Public Module Gibberish
	Private ReadOnly Random As Random = New Random()

	''' <summary>
	''' Generate gibberish.
	''' </summary>
	''' <param name="size">The size of the gibberish.</param>
	''' <returns>An <see cref="Array"/> of UTF-8 gibberish.</returns>
	Public Function GenerateUTF8(ByVal size As Int32) As Byte()
		Dim result = New Byte(size) {}
		Random.NextBytes(result)
		Return result
	End Function

	''' <summary>
	''' Generate gibberish.
	''' </summary>
	''' <param name="size">The size of the gibberish.</param>
	''' <returns>An <see cref="Array"/> of UTF-16 gibberish.</returns>
	Public Function GenerateUTF16(ByVal size As Int32) As Char()
		Dim result = New Char(size) {}
		For i = 0 To size
			result(i) = ChrW(Random.Next() Mod UInt16.MaxValue)
		Next
		Return result
	End Function

	''' <summary>
	''' Generate gibberish.
	''' </summary>
	''' <param name="size">The size of the gibberish.</param>
	''' <returns>An <see cref="Array"/> of UTF-32 gibberish.</returns>
	Public Function GenerateUTF32(ByVal size As Int32) As Rune()
		Dim result = New Rune(size) {}
		For i = 0 To size
			Dim value = Random.Next()
			If value >= &H10FFFF Then
				value = value Mod &H10FFFF
			End If
			If &HD800 <= value AndAlso value <= &HDFFF Then
				value -= &HD800
			End If
			result(i) = New Rune(value)
		Next
		Return result
	End Function
End Module
